scripts/addons.py

```diff
@@ -0,0 +1,249 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Elico Corp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import os
+import re
+from os import remove
+from shutil import move
+from urlparse import urlparse
+from subprocess import call
+from subprocess import check_output
+
+EXTRA_ADDONS_PATH = '/mnt/data/additional_addons/'
+OLD_ODOO_CONF = '/opt/odoo/etc/odoo.conf.old'
+ODOO_CONF = '/opt/odoo/etc/odoo.conf'
+ADDONS_PATH = ['/opt/odoo/sources/odoo/addons']
+DEFAULT_SCHEME = 'https://'
+DEFAULT_GIT_REPO_HOSTING_SERVICE = 'github.com'
+DEFAULT_ORGANIZATION = 'OCA'
+DEPENDENCIES_FILE = 'oca_dependencies.txt'
+REGEX_ADDONS_PATH = r'^addons_path\s*=\s*'
+
+
+class Repo(object):
+    """
+    oca_dependencies.txt
+
+    For public repo:
+
+    oca-repo -> https://github.com/OCA/oca-repo (default branch)
+    oca-repo 8.0 -> https://github.com/OCA/oca-repo (branch: 8.0)
+    organization/public-repo -> https://github.com/organization/public-repo (default branch)
+    organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+    https://github.com/organization/public-repo -> https://github.com/organization/public-repo (default branch)
+    https://github.com/organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+    public_repo_rename https://github.com/organization/public-repo -> https://github.com/organization/public-repo (default branch)
+    public_repo_rename https://github.com/organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+
+    For private repo:
+
+    git@github.com:Elico-Corp/private-repo
+    git@github.com:Elico-Corp/private-repo 8.0
+    private_repo_rename git@github.com:Elico-Corp/private-repo
+    private_repo_rename git@github.com:Elico-Corp/private-repo 8.0
+
```

Comment:
I understand. I think it is not so big deal but I agree that it takes a lot of time and resources. Maybe you can think about an alternative added to current method: we could fetch the zip modules from apps.odoo.com (it should not be difficult tp build a fetch function based on the name of module and branch) and unzip in specific folder.
Nevertheless, the current name in apps is random (probably for this reason ;)): https://apps.odoo.com/loempia/download/oerp_no_phoning_home/9.0.9.02/7eBOBea88gYyLOur3rzx9L.zip?deps
Then a simple option is to provide that URL inside the dependency file (provided they dont change often...)
Currently I think git doesnot allow fetching one module but not not sure about all possible API features from Github.